x/staking/keeper/grpc_query.go

```diff
@@ -0,0 +1,417 @@
+package keeper
+
+import (
+	"context"
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+type Querier struct {
+	Keeper
+}
+
+var _ types.QueryServer = Querier{}
+
+func (k Querier) Validators(c context.Context, req *types.QueryValidatorsRequest) (*types.QueryValidatorsResponse, error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
+	}
+
+	if req.Status == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid request")
+	}
+
+	var validators types.Validators
+	ctx := sdk.UnwrapSDKContext(c)
+
+	store := ctx.KVStore(k.storeKey)
+	valStore := prefix.NewStore(store, types.ValidatorsKey)
+
+	res, err := query.FilteredPaginate(valStore, req.Req, func(key []byte, value []byte, accumulate bool) (bool, error) {
+		val, err := types.UnmarshalValidator(k.cdc, value)
+		if err != nil {
+			return false, err
+		}
+
+		if strings.EqualFold(val.GetStatus().String(), req.Status) {
```

Comment:
curious why are you using EqualFold instead of == ?

Reply:
I remember seeing benchmarks for this once, and equalfold is most performant . It shouldn't matter much here though